pkg/workflows/upgrade.go

```diff
@@ -280,7 +276,7 @@ func (s *pauseEksaAndFluxReconcile) Name() string {
 }
 
 func (s *createBootstrapClusterTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
-	if commandContext.BootstrapCluster != nil && commandContext.BootstrapCluster.ExistingManagement {
+	if commandContext.ManagementCluster != nil && commandContext.ManagementCluster.ExistingManagement {
```

Comment:
it might be cleaner if we use `!commandContext.WorkloadCluster.IsSelfManaged()`?

Reply:
`IsSelfManaged` isn't a method we can call here because this Cluster object is different from the cluster type object from the v1alpha1 pkg.